server/bundles/io.cloudbeaver.service.admin/src/io/cloudbeaver/service/admin/impl/WebServiceAdmin.java

```diff
@@ -66,16 +69,38 @@ public class WebServiceAdmin implements DBWServiceAdmin {
         .flatMap(service -> service.getPermissions().stream())
         .collect(Collectors.toMap(WebPermissionDescriptor::getId, Function.identity()));
 
+    @NotNull
+    @Override
+    public AdminUserInfo getUserById(@NotNull WebSession webSession, @NotNull String userId) throws DBWebException {
+        try {
+            SMUser smUser = webSession.getAdminSecurityController().getUserById(userId);
+            return new AdminUserInfo(webSession, new WebUser(smUser));
+        } catch (Exception e) {
+            throw new DBWebException("Error getting user - " + userId, e);
+        }
+    }
 
     @NotNull
     @Override
-    public List<AdminUserInfo> listUsers(@NotNull WebSession webSession, String userName) throws DBWebException {
+    public GQLConnection<AdminUserInfo> listUsers(@NotNull WebSession webSession, Integer first, String after,
+            String userIdMask, Boolean enabledState) throws DBWebException {
         try {
-            List<AdminUserInfo> webUsers = new ArrayList<>();
-            for (SMUser smUser : webSession.getAdminSecurityController().findUsers(userName)) {
-                webUsers.add(new AdminUserInfo(webSession, new WebUser(smUser)));
+            List<GQLConnectionEdge<AdminUserInfo>> edges = new ArrayList<>();
+            for (SMUser smUser : webSession.getAdminSecurityController().findUsers(first, after, userIdMask, enabledState)) {
+                edges.add(
+                        new GQLConnectionEdge<>(new AdminUserInfo(webSession, new WebUser(smUser)),
+                                smUser.getUserId()));
             }
-            return webUsers;
+            int totalCount = webSession.getAdminSecurityController().countUsers(null, userIdMask, enabledState);
```

Comment:
we can return search results with already known count, and we don't need to additionally call the count method

Reply:
removed